### ISSUE TITLE
can't send message without any error

### ISSUE THREAD

**Issue by LeeWongClient** [bridgefy/bridgefy_flutter#36]
Hello, I integrated bridgefy’s flutterSDK

`dependencies:
 bridgefy: ^1.1.5`

I downloaded the demo project (https://github.com/bridgefy/bridgefy_flutter) and replaced the application id and api key of the project. The referenced program can init and start normally.

But when I try to send a message (at least one device around is open Android 10+), but
Message sender: Did not receive any callback regarding the success or failure of message sending.

Message receiver: No message received and no callback information
may I know what is the reason?

I didn't modify any demo code just application id and api key  that mentioned above

Here is my log

```
Debug service listening on ws://127.0.0.1:62967/UJh0SNeTHUc=/ws
Syncing files to device PBEM00...
W/Gralloc3(15018): mapper 3.x is not supported
D/SurfaceView(15018): 202177657 updateSurfacePosition RenderWorker, frameNr = 1, postion = [0, 0, 1080, 2340]
D/ProfileInstaller(15018): Installing profile for com.example.bledemo
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->objectFieldOffset(Ljava/lang/reflect/Field;)J (greylist,core-platform-api, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putLong(Ljava/lang/Object;JJ)V (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->allocateInstance(Ljava/lang/Class;)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekLong(JZ)J (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeLong(JJZ)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeInt(JIZ)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekInt(JZ)I (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeByte(JB)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekByte(J)B (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeByteArray(J[BII)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekByteArray(J[BII)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->arrayBaseOffset(Ljava/lang/Class;)I (greylist,core-platform-api, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->arrayIndexScale(Ljava/lang/Class;)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putLong(Ljava/lang/Object;JJ)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden field Ljava/nio/Buffer;->address:J (greylist, reflection, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
I/chatty  (15018): uid=10337(com.example.bledemo) identical 8 lines
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
I/chatty  (15018): uid=10337(com.example.bledemo) identical 3 lines
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
D/NetworkSecurityConfig(15018): No Network Security Config specified, using platform default
W/example.bledem(15018): Accessing hidden method Ljava/lang/invoke/MethodHandles$Lookup;-><init>(Ljava/lang/Class;I)V (greylist, reflection, allowed)
D/Bridgefy(15018): Bridgefy was initialized successful with a7d9b264-b950-475b-96c6-a7186c576e56
I/Bridgefy(15018): Bridgefy license valid to Mon May 27 09:44:32 GMT+08:00 2024
E/CheckPermission(15018): _bluetooth code = 1 
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothGattServer(15018): registerCallback()
D/BluetoothGattServer(15018): registerCallback() - UUID=d8d80a54-ccfa-4e0b-9b3a-a8481323086d
D/BluetoothGattServer(15018): onServerRegistered() - status=0 serverIf=6
D/BluetoothGattServer(15018): addService() - service: 2bee975d-eca3-21f1-5359-aa63810e2025
I/Bridgefy(15018): Bridgefy started with a7d9b264-b950-475b-96c6-a7186c576e56 client 9aafbc40-cdef-495f-ac11-2caff74f4f79
D/BluetoothGattServer(15018): onServiceAdded() - handle=43 uuid=2bee975d-eca3-21f1-5359-aa63810e2025 status=0
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothLeScanner(15018): onScannerRegistered() - status=0 scannerId=8 mScannerId=0
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
I/Choreographer(15018): Skipped 1 frames!  The application may be doing too much work on its main thread.
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
I/Choreographer(15018): Skipped 1 frames!  The application may be doing too much work on its main thread.
E/CheckPermission(15018): _bluetooth code = 10 
D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
D/BluetoothAdapter(15018): isLeEnabled(): ON
E/CheckPermission(15018): _bluetooth code = 10 
E/CheckPermission(15018): _bluetooth code = 10 
```

**Comment by LeeWongClient** [comment]
@julian-bridgefy  @rafaelcr @kekoyde @ghbfy

**Comment by LeeWongClient** [comment]
Android： 
when click start  app crash 
I/QarthLog(32626): [PatchStore] createDisableExceptionQarthFile
I/QarthLog(32626): [PatchStore] create disable file for com.example.bledemo uid is 10290
E/AndroidRuntime(32626): FATAL EXCEPTION: main
E/AndroidRuntime(32626): Process: com.example.bledemo, PID: 32626
E/AndroidRuntime(32626): java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
E/AndroidRuntime(32626): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:523)
E/AndroidRuntime(32626): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)
E/AndroidRuntime(32626): Caused by: java.lang.reflect.InvocationTargetException
E/AndroidRuntime(32626): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(32626): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
E/AndroidRuntime(32626): 	... 1 more
E/AndroidRuntime(32626): Caused by: UnknownException(error=java.nio.BufferUnderflowException)
I/Process (32626): Sending signal. PID: 32626 SIG: 9

**Comment by LeeWongClient** [comment]
> Hello, I integrated bridgefy’s flutterSDK
> 
> `dependencies: bridgefy: ^1.1.5`
> 
> I downloaded the demo project (https://github.com/bridgefy/bridgefy_flutter) and replaced the application id and api key of the project. The referenced program can init and start normally.
> 
> But when I try to send a message (at least one device around is open Android 10+), but Message sender: Did not receive any callback regarding the success or failure of message sending.
> 
> Message receiver: No message received and no callback information may I know what is the reason?
> 
> I didn't modify any demo code just application id and api key that mentioned above
> 
> Here is my log
> 
> ```
> Debug service listening on ws://127.0.0.1:62967/UJh0SNeTHUc=/ws
> Syncing files to device PBEM00...
> W/Gralloc3(15018): mapper 3.x is not supported
> D/SurfaceView(15018): 202177657 updateSurfacePosition RenderWorker, frameNr = 1, postion = [0, 0, 1080, 2340]
> D/ProfileInstaller(15018): Installing profile for com.example.bledemo
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->objectFieldOffset(Ljava/lang/reflect/Field;)J (greylist,core-platform-api, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putLong(Ljava/lang/Object;JJ)V (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->allocateInstance(Ljava/lang/Class;)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekLong(JZ)J (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeLong(JJZ)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeInt(JIZ)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekInt(JZ)I (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeByte(JB)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekByte(J)B (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->pokeByteArray(J[BII)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Llibcore/io/Memory;->peekByteArray(J[BII)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->arrayBaseOffset(Ljava/lang/Class;)I (greylist,core-platform-api, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->arrayIndexScale(Ljava/lang/Class;)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putLong(Ljava/lang/Object;JJ)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden field Ljava/nio/Buffer;->address:J (greylist, reflection, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> I/chatty  (15018): uid=10337(com.example.bledemo) identical 8 lines
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> I/chatty  (15018): uid=10337(com.example.bledemo) identical 3 lines
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
> W/example.bledem(15018): Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
> D/NetworkSecurityConfig(15018): No Network Security Config specified, using platform default
> W/example.bledem(15018): Accessing hidden method Ljava/lang/invoke/MethodHandles$Lookup;-><init>(Ljava/lang/Class;I)V (greylist, reflection, allowed)
> D/Bridgefy(15018): Bridgefy was initialized successful with a7d9b264-b950-475b-96c6-a7186c576e56
> I/Bridgefy(15018): Bridgefy license valid to Mon May 27 09:44:32 GMT+08:00 2024
> E/CheckPermission(15018): _bluetooth code = 1 
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothGattServer(15018): registerCallback()
> D/BluetoothGattServer(15018): registerCallback() - UUID=d8d80a54-ccfa-4e0b-9b3a-a8481323086d
> D/BluetoothGattServer(15018): onServerRegistered() - status=0 serverIf=6
> D/BluetoothGattServer(15018): addService() - service: 2bee975d-eca3-21f1-5359-aa63810e2025
> I/Bridgefy(15018): Bridgefy started with a7d9b264-b950-475b-96c6-a7186c576e56 client 9aafbc40-cdef-495f-ac11-2caff74f4f79
> D/BluetoothGattServer(15018): onServiceAdded() - handle=43 uuid=2bee975d-eca3-21f1-5359-aa63810e2025 status=0
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothLeScanner(15018): onScannerRegistered() - status=0 scannerId=8 mScannerId=0
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> I/Choreographer(15018): Skipped 1 frames!  The application may be doing too much work on its main thread.
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> I/Choreographer(15018): Skipped 1 frames!  The application may be doing too much work on its main thread.
> E/CheckPermission(15018): _bluetooth code = 10 
> D/BluetoothAdapter(15018): 111160112: getState(). Returning ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> D/BluetoothAdapter(15018): isLeEnabled(): ON
> E/CheckPermission(15018): _bluetooth code = 10 
> E/CheckPermission(15018): _bluetooth code = 10 
> ```

After I restarted the device, this scenario was no longer a problem, but I'm not sure why.

**Comment by LeeWongClient** [comment]
> Android： when click start, app crash
> 
> ```
> I/QarthLog(32626): [PatchStore] createDisableExceptionQarthFile
> I/QarthLog(32626): [PatchStore] create disable file for com.example.bledemo uid is 10290
> E/AndroidRuntime(32626): FATAL EXCEPTION: main
> E/AndroidRuntime(32626): Process: com.example.bledemo, PID: 32626
> E/AndroidRuntime(32626): java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
> E/AndroidRuntime(32626): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:523)
> E/AndroidRuntime(32626): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)
> E/AndroidRuntime(32626): Caused by: java.lang.reflect.InvocationTargetException
> E/AndroidRuntime(32626): 	at java.lang.reflect.Method.invoke(Native Method)
> E/AndroidRuntime(32626): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
> E/AndroidRuntime(32626): 	... 1 more
> E/AndroidRuntime(32626): Caused by: UnknownException(error=java.nio.BufferUnderflowException)
> I/Process (32626): Sending signal. PID: 32626 SIG: 9
> ```

The scenario to reproduce this problem is:
1. Open the iOS APP and put it next to it
2. This problem will occur when the start method is called after the Android APP init is successfully opened. I wonder if there is a problem in the process of establishing a connection between iOS and Android.
But the code should be in the library, so please confirm it

**Comment by julian-bridgefy** [comment]
Hi @LeeWongClient,

Thank you for reporting this issue and providing detailed steps to reproduce it.

We have identified that the issue lies within the iOS-Android connectivity process. Our team has been actively working on resolving this. I'm pleased to inform you that we have recently released a new version of the Bridgefy iOS SDK, which includes several connectivity improvements. We are also in the final stages of releasing an updated version of the Bridgefy Android SDK, which will address similar connectivity issues. Both updates are part of our ongoing efforts to enhance cross-platform compatibility.

We appreciate your patience and understanding as we work to resolve this issue.
